homework2/StackCalculator.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Homework2;
+/// <summary>
+/// Represents a calculator that works with reverse polish notation.
+/// </summary>
+public class StackCalculator
+{
+    private IStack _stack;
+    
+    public StackCalculator(IStack stack)
+    {
+        _stack = stack;
+    }
+    
+    public double Calculate(string expression)
+    {
+        var input = expression.Split();
+        double result = 0;
+        foreach (var element in input)
+        {
+            if (int.TryParse(element, out var number))
+            {
+                _stack.Push(number);
+            }
+            else
+            {
+                // тут по идее бросится исключение и поймается в мейне
+                var storedNumber = _stack.Pop();
+                var inputNumber = _stack.Pop();
+                if (element == "/" && Math.Abs(inputNumber) < Math.Pow(10, -9))
+                {
+                    throw new DivideByZeroException();
+                }
+                result = element switch
+                {
+                    "+" => storedNumber + inputNumber,
+                    "-" => storedNumber - inputNumber,
+                    "*"=> storedNumber * inputNumber,
```

Comment:
```suggestion
                    "*" => storedNumber * inputNumber,
```
Можно считаться настоящим программистом, когда пропущенный пробел в коде доставляет почти физические страдания :)